File: eco-core/core-proxy/src/main/java/com/willfp/eco/proxy/VillagerTradeProxy.java
package com.willfp.eco.proxy;

import com.willfp.eco.core.proxy.AbstractProxy;
import org.bukkit.entity.Player;
import org.bukkit.inventory.MerchantRecipe;
import org.jetbrains.annotations.NotNull;

public interface VillagerTradeProxy extends AbstractProxy {
    MerchantRecipe displayTrade(@NotNull MerchantRecipe recipe,
                                @NotNull Player player);
}

File: eco-core/core-proxy/src/main/kotlin/com/willfp/eco/proxy/VillagerTradeProxy.kt
package com.willfp.eco.proxy

import com.willfp.eco.core.proxy.AbstractProxy
import org.bukkit.entity.Player
import org.bukkit.inventory.MerchantRecipe

interface VillagerTradeProxy : AbstractProxy {
    fun displayTrade(
        recipe: MerchantRecipe,
        player: Player
    ): MerchantRecipe
}